From 5158ebce305d961a2d2e3cb3f889b0cde593c4a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Hole=C4=8Dek?= <oholecek@aaannz.eu>
Date: Mon, 10 May 2021 16:23:19 +0200
Subject: [PATCH] Fix missing minion returns in batch mode (#360)

Don't close pub if there are pending events, otherwise events will be lost
resulting in empty minion returns.

Co-authored-by: Denis V. Meltsaykin <dmeltsaykin@mirantis.com>
---
 salt/client/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/salt/client/__init__.py b/salt/client/__init__.py
index bcda56c9b4..b2617e4554 100644
--- a/salt/client/__init__.py
+++ b/salt/client/__init__.py
@@ -976,7 +976,7 @@ class LocalClient:
 
                 self._clean_up_subscriptions(pub_data["jid"])
         finally:
-            if not was_listening:
+            if not was_listening and not self.event.pending_events:
                 self.event.close_pub()
 
     def cmd_full_return(
-- 
2.39.2


